src/us/kbase/auth2/lib/storage/mongo/MongoStorage.java

```diff
@@ -748,33 +748,49 @@ private Document getUserDoc(
 			final boolean local)
 			throws AuthStorageException, NoSuchUserException {
 		requireNonNull(userName, "userName");
-		Document user = getUserDocWithoutPassword(collection, userName);
+		return getUserDoc(
+				collection, new Document(Fields.USER_NAME, userName.getName()), local, userName);
+		
+	}
+	
+	private Document getUserDoc(
+			final String collection,
+			final Document query,
+			final boolean local,
+			final UserName nameForException)
```

Comment:
Does anything else call this version of the method? Should it also `requireNonNull(username)` at the top? If it's only getting called by the other method, that's fine.

Reply:
It's called by the other `getUserDoc` method as well as `getUser(RemoteIdentity)`. There can't be a null check since `nameForException` is allowed to be null, as that signifies an exception shouldn't be thrown, which is what `getUser(RemoteIdentity)` expects